### PR TITLE
feat: address management

### PR DESCRIPTION
**Pull request by bhtibrewal**
added address management with backend
Pages to check: 
- [Address Controller](https://github.com/bhtibrewal/ArtsyShop/compare/address-management?expand=1#diff-66dccb0b3654aca5ffc84201bf3a2b252c62f201059e9fea62ef1cf6ace7ddee)
- [AddressForm.jsx](https://github.com/bhtibrewal/ArtsyShop/compare/address-management?expand=1#diff-5330c62656e85271c6976f9de50ca9a9eca6a03087b8e6837db564641fed9b18)
- [AddressesList.jsx](https://github.com/bhtibrewal/ArtsyShop/compare/address-management?expand=1#diff-4e9c24ac2e7888eef3fd7d4ac2c79badef565476a88ce993a46ab86b94b14a5c)

I would appreciate if you could login and check the ui/ux as well

### CODE REVIEW

**Review by Naman-Saxena1 — CHANGES_REQUESTED** [review]
Code quality, variable naming conventions, UI design are good.
Addresses can be added and removed for logged in user, functionality is also working great 👍 

1 UI suggestion, toast's z-index is lower than wishlist button and higher than Navbar, z-index and placement of toast can be improved to make sure it does not hide navbar and does not go below wishlist button.


Capture4

File: src/backend/controllers/AddressController.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    } catch (error) {
+        return new Response(
+            500,
+            {},
+            {
+                error,
+            }
+        );
+    }
+};
```
Comment (Ankur9669) [inline src/backend/controllers/AddressController.js:74]: Appreciation: Very Nice that you have created custom Api

File: src/pages/checkout/components/AddressForm.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { addNewAddress } from "../../../services";
+
+export const AddressForm = () => {
+  const {
+    userData: { addresses },
+    userDataDispatch,
+  } = useUserContext();
+  const { showToast } = useToast();
+  const addressEntries = { street: "", city: "", country: "", zip_code: "" };
+  const [inputValues, setInputValues] = useState({ ...addressEntries });
```
Comment (Ankur9669) [inline src/pages/checkout/components/AddressForm.jsx:13]: Awesome use of addressEntries object

File: src/pages/checkout/components/AddressForm.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  const submitAddressFormHandler = (e) => {
+    e.preventDefault();
+    if (Object.keys(inputValues).some((element) => inputValues[element] === ""))
```
Comment (Ankur9669) [inline src/pages/checkout/components/AddressForm.jsx:17]: What's Happening here?

**Review by Ankur9669 — APPROVED** [review]
THe code looks really nice, great work!!!

File: src/pages/checkout/components/AddressForm.jsx (hunk 3 above)
Comment (bhtibrewal) [reply src/pages/checkout/components/AddressForm.jsx:17]: Checking if none of the entries of the form is empty

**Review by yash9263 — APPROVED** [review]
Everything looks good. The code is well written. Good use of objects in useState for updating states.

**Pull request merged**